### ISSUE TITLE
problem with bind-interfaces in /etc/dnsmasq.conf

### ISSUE THREAD

**Issue by ghost** [arpitjindal97/raspbian-recipes#6]
Ok I'm beating my head against the wall with this one so I figured I'd post an issue to see if I can figure it out.

I've used your script in the past and everything worked great,  it made it a lot easier than doing everything manually.

However I needed to make up a new sd card and I hadn't backed up....  So I'm starting from the beginning again unfortunately.

I'm using the aiyprojects-2018-2-21 img which is a google aiy kit version of raspbian stretch.
dnsmasq is already installed,  and the system is fully updated.

When your script runs I get Job for dnsmasq.service failed because the control process exited with error code.

checking systemctl status dnsmasq shows:

```
pi@raspberrypi:~ $ systemctl status dnsmasq
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
   Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2018-04-13 08:28:53 EST; 14s ago
  Process: 4695 ExecStop=/etc/init.d/dnsmasq systemd-stop-resolvconf (code=exited, status=0/SUCCESS)
  Process: 1171 ExecStartPost=/etc/init.d/dnsmasq systemd-start-resolvconf (code=exited, status=0/SUCCESS)
  Process: 4760 ExecStart=/etc/init.d/dnsmasq systemd-exec (code=exited, status=2)
  Process: 4757 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)
 Main PID: 1170 (code=exited, status=0/SUCCESS)

Apr 13 08:28:53 raspberrypi systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
Apr 13 08:28:53 raspberrypi dnsmasq[4757]: dnsmasq: syntax check OK.
Apr 13 08:28:53 raspberrypi dnsmasq[4760]: dnsmasq: unknown interface usb0
Apr 13 08:28:53 raspberrypi systemd[1]: dnsmasq.service: Control process exited, code=exited status=2
Apr 13 08:28:53 raspberrypi systemd[1]: Failed to start dnsmasq - A lightweight DHCP and caching DNS server.
Apr 13 08:28:53 raspberrypi systemd[1]: dnsmasq.service: Unit entered failed state.
Apr 13 08:28:53 raspberrypi systemd[1]: dnsmasq.service: Failed with result 'exit-code'.
```

If I remove the bind-interfaces command from /etc/dnsmasq.conf,  dnsmasq will start so I know the problem has something to do with that line.  (but it doesn't actually bind the interfaces)

Now,  my first thought was that I had something wrong with my config and that I needed to edit your script to use usb0 for something.  (Your script is unchanged at this point)

However ifconfig produces:

```
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.2.1  netmask 255.255.255.0  broadcast 192.168.2.255
        inet6 fe80::a99c:7101:9e08:45b1  prefixlen 64  scopeid 0x20<link>
        ether b8:27:eb:03:81:34  txqueuelen 1000  (Ethernet)
        RX packets 142  bytes 28674 (28.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 163  bytes 31199 (30.4 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 20  bytes 1557 (1.5 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 20  bytes 1557 (1.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.0.93  netmask 255.255.255.0  broadcast 10.0.0.255
        inet6 2601:40e:4000:cdc7::ce6c  prefixlen 128  scopeid 0x0<global>
        inet6 fe80::4e91:97fb:4096:81aa  prefixlen 64  scopeid 0x20<link>
        inet6 2601:40e:4000:cdc7:60e5:f552:85c5:e62d  prefixlen 64  scopeid 0x0<global>
        ether b8:27:eb:56:d4:61  txqueuelen 1000  (Ethernet)
        RX packets 1172  bytes 145291 (141.8 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 372  bytes 63341 (61.8 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

So I'm really unsure why it's trying to do something with usb0

**Comment by arpitjindal97** [comment]
If I am guessing right than you are using `wifi-to-eth-route.sh` script. Looking at the logs of `dnsmasq` tells that you have edited the script and replaced `eth0` with `usb0`. 

If you have nothing to do with usb interface than undo any changes you made to script or download it again. Also explain a little bit about how to want to route the traffic.

Feel free to ask anything if it doesn't work.

**Comment by ghost** [comment]
Yeah,  I'm sorry I am using wifi-to-eth-route.sh,  I'm just using it to route from the wifi to ethernet so that an old computer that doesn't have a working wifi card anymore can use it to get on my home network.  

As far as me editing your script,  I haven't.  That's why I'm scratching my head.  Attached is 
[wifi-to-eth-route.sh.zip](https://github.com/arpitjindal97/raspbian-recipes/files/1908518/wifi-to-eth-route.sh.zip)

the copy of your script that I'm using.  I was hoping you would have some idea where usb0 is coming from.
(note:  I had to zip it to upload it in the messages here.  As you can see,  it is unchanged.)

**Comment by arpitjindal97** [comment]
Try this workaround to debug the issue 

- Remove script command from autostart, let pi boot normally
- Attach monitor, keyboard to it and run the script normally
- Open another terminal, and check if dnsmasq is running or not.
- If it is not then open the file and see if it contains `eth0` or `usb0`
- if it is `usb0` the edit it and replace it with `eth0`
- Start the dnsmasq service manually
- check again the status of dnsmasq if it failed or running

If it is running then restart the pi and start script again manually and check if everything is fine or not.

If still it is not working then you should try it in official raspbian image.

**Comment by ghost** [comment]
Yeah works on official raspbian.  Just doesn't on the AIY-images.

**Comment by arpitjindal97** [comment]
Google's official [website](https://aiyprojects.withgoogle.com/vision/) says AIY image is for Raspberry Pi Zero. The device doesn't have ethernet port and maybe you are trying this image on RPi 3. So, dnsmasq which is already installed is searching for usb0 (hardcoded). 

Give me some time to try AIY image on RPi 3

**Comment by ghost** [comment]
Double checking myself that I pulled the right image,  as I'm using the image from here:  https://aiyprojects.withgoogle.com/voice/

The voice kit can work with the 0 with limited functionality but is meant to be on a 3.

**Comment by arpitjindal97** [comment]
So, I will try AIY voice image and see if I can do something :thinking:

**Comment by ghost** [comment]
Yeah all the images are the same.  I ran shasum on the image I was using,  one downloaded from the voice page,  and one downloaded from the vision page.  All three images hash the same so they're the same image regardless of what project you are working on.  (which makes sense in a way,  I have the source code for the vision project and the voice project on the image)

**Comment by arpitjindal97** [comment]
Solved the problem, turns out to be google devs have put `/etc/dnsmasq.d/googadget-dnsmasq.conf` file.

Dnsmasq first reads `/etc/dnsmasq.conf` and then reads all the config files in `/etc/dnsmasq.d/` directory. So, `googadget-dnsmasq.conf` file was overriding the configurations.

I have made a change in script now. Config will now be stored in `/etc/dnsmasq.d` directory and all other files will be removed from there.  

Try the new script and see if it works
